AK/Atomic.h

```diff
@@ -187,6 +187,7 @@ class Atomic {
         return *ret;
     }
 
+    // NOLINTNEXTLINE(misc-unconventional-assign-operator) Returning an object of type Atomic volatile& here is a compilation error
```

Comment:
Here and the other two in this file:
`volatile` marks the method as "can be called on instances of this class, as long as the instance can be converted to volatile Atomic<T>". That is always the case, so I don't understand what `volatile` is doing here. (Really, I don't understand. It might be correct, it might be incorrect.)

My point is: It doesn't refer to the return type. The comment is therefore misleading.

I agree that we want to return a `T` here, and not an `Atomic<T>`.

Suggestion: Change the comment to say something like `We want operator= to exchange the value, so returning an object of type Atomic& here does not make sense.`

Reply:
It doesn't compile with Atomic& on the first pass of the compiler. Because it's a volatile method, it would need to be Atomic volatile &. But that construct doesn't compile at the point of use (in tests). But yes, I agree that the statement of why the written behavior is correct is better than explaining a compile error 😅